docs/sequence_diagrams/registration.md

```diff
@@ -0,0 +1,203 @@
+    participant cc_ipr as CC Intel Platform Registration
+    participant pcs as SGX Platform Certification Service (PCS)
+    participant pce as SGX Provisioning Certification Enclave (PCE)
+
+    autonumber
+
+    activate cc_ipr
+
+    cc_ipr->>cc_ipr: Read UEFI variable SgxRegistrationStatus
+
+    opt UEFI variable SgxRegistrationStatus does NOT exist
+        cc_ipr->>cc_ipr: Return status code 02
+    end
+
+    alt Flag SgxRegistrationStatus.SgxRegistrationComplete is UNSET 
+        cc_ipr->>cc_ipr: Read UEFI variable SgxRegistrationServerRequest
+        note right of cc_ipr: The Platform Manifest is available in that variable
+
+        opt UEFI variable SgxRegistrationServerRequest does NOT exist
+            cc_ipr->>cc_ipr: Return status code 02
+        end
+
+        cc_ipr->>cc_ipr: Register platform(Platform Manifest)
+        note right of cc_ipr: See diagram `2.1. Registration`
+
+        cc_ipr->>cc_ipr: Return status code
+
+    else Flag SgxRegistrationStatus.SgxRegistrationComplete is SET
+        opt UEFI variable SgxRegistrationServerRequest exists
+            note right of cc_ipr: We finished the registration and updated SgxRegistrationStatus.SgxRegistrationComplete.<br> However, the reboot has not been performed yet so that<br> the BIOS can remove SgxRegistrationServerRequest
+            cc_ipr->>cc_ipr: Return status code 05
+        end
+
+        note right of cc_ipr: We want to determine whether Direct or Indirect Registration was performed
+
+        cc_ipr->>+pce: Read the PCE info 
+            note right of cc_ipr: We use the Intel SGX SDK to manage the communication with the SGX PCE (`sgx_get_pce_info`)
+        pce-->>-cc_ipr: Encrypted PPID, PCEID
+
+        opt Error while retrieving PCE info
+            cc_ipr->>cc_ipr: Return status code 03
+        end
+
+        cc_ipr->>+pcs: GET https://api.trustedservices.intel.com/sgx/certification/v4/pckcerts (body: Encrypted PPID, PCEID)
+            note right of cc_ipr: Returns the PCK Cert if the Intel RS has cached the platform root keys (aka. Direct Registration)
+        pcs-->>-cc_ipr: JSON data structure containing a collection of PCK Certs
+
+        alt HTTP Status Code 200
+            cc_ipr->>cc_ipr: Return status code 00
+        else HTTP Status Code 404
+            note right of cc_ipr: Registration set as completed but We canNOT determine if an indirect registration has been carried out,<br> 404 might happen because:<br> (i) the direct registration failed, or<br> (ii) the indirect registration was performed
+            cc_ipr->>cc_ipr: Return status code 04
+        else
+            note right of cc_ipr: The data available at this moment are insufficient to determine what to do next
+            cc_ipr->>cc_ipr: Return status code 03
+        end     
+    end
+
+    deactivate cc_ipr
+```
+
+#### 2.1. Registration
+
+To set the `Key Caching Policy` to true, we **must** register the Platform with Intel Registration Service first.
```

Comment:
The start with the `Key Caching Policy` could be confusing. I suggest to start first with explaining what we understand under *Registration* (uploading the retrieved platfrom manifest to the Intel Registration Service).